homeassistant/components/madvr/config_flow.py

```diff
@@ -0,0 +1,161 @@
+"""Config flow for the integration."""
+
+import asyncio
+import logging
+
+import aiohttp
+from madvr.madvr import HeartBeatError, Madvr
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.config_entries import ConfigEntry, ConfigFlowResult, OptionsFlow
+from homeassistant.const import CONF_HOST, CONF_MAC, CONF_NAME, CONF_PORT
+from homeassistant.core import callback
+
+from .const import DOMAIN
+from .coordinator import MadVRCoordinator
+from .utils import CannotConnect, cancel_tasks
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class MadVRConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for the integration."""
+
+    async def async_step_user(self, user_input=None) -> ConfigFlowResult:
+        """Handle the initial step."""
+        errors = {}
+
+        if user_input is not None:
+            host = user_input[CONF_HOST]
+            port = user_input.get(CONF_PORT, 44077)
+            mac = user_input[CONF_MAC]
+            keep_power_on = user_input["keep_power_on"]
```

Comment:
What does this setting mean

Reply:
During adding the device, it has to be on to be connected. If the user wishes, the integration can turn it off after or keep it on

Reply:
moved to a const also